src/ossl/ecdh.rs

```diff
@@ -28,7 +28,13 @@ use pkcs11::*;
 /// ossl DigestAlg
 fn kdf_type_to_digest_alg(mech: CK_EC_KDF_TYPE) -> Result<DigestAlg> {
     Ok(match mech {
-        CKD_SHA1_KDF | CKD_SHA1_KDF_SP800 => DigestAlg::Sha1,
+        CKD_SHA1_KDF => {
+            if cfg!(feature = "no_sha1") {
+                return Err(CKR_MECHANISM_PARAM_INVALID)?;
+            } else {
+                return Ok(DigestAlg::Sha1);
+            }
+        }
```

Comment:
My point was to adjust the code similarly also in the other cases that have the same pattern, so here
```suggestion
        #[cfg(not(feature = "no_sha1"))]
        CKD_SHA1_KDF => {
            return Ok(DigestAlg::Sha1);
        }
```
and accordingly in the other files that have the same pattern.